samplers/samplers.go

```diff
@@ -317,16 +317,19 @@ func (h *Histo) Flush(interval time.Duration, percentiles []float64, aggregates
 			Tags:       tags,
 			MetricType: "gauge",
 		})
-		if (aggregates.Value&AggregateAverage) == AggregateAverage && h.LocalWeight != 0 {
-			// we need both a rate and a non-zero sum before it will make sense
-			// to submit an average
-			metrics = append(metrics, DDMetric{
-				Name:       fmt.Sprintf("%s.avg", h.Name),
-				Value:      [1][2]float64{{now, h.LocalSum / h.LocalWeight}},
-				Tags:       tags,
-				MetricType: "gauge",
-			})
-		}
+	}
+
+	if (aggregates.Value&AggregateAverage) == AggregateAverage && h.LocalSum != 0 && h.LocalWeight != 0 {
+		// we need both a rate and a non-zero sum before it will make sense
+		// to submit an average
+		tags := make([]string, len(h.Tags))
```

Comment:
I'm definitely fine doing this, but as I was doing it I noticed this comment on the first instance of it:
```go
// Defensively recopy tags to avoid aliasing bugs in case multiple DDMetrics share the same
// tag array in the future
```
Which I don't quite understand the problem its trying to solve, can you confirm that the comment isn't something we want to worry about and I can go ahead an make this change